packages/babel-plugin-transform-merge-sibling-variables/src/index.js

```diff
@@ -1,9 +1,51 @@
 "use strict";
 
-module.exports = function() {
+module.exports = function({ types: t }) {
   return {
     name: "transform-merge-sibling-variables",
     visitor: {
+      ForStatement(path) {
+        // Lift declarations to the loop initializer
+        let body = path.get("body");
+        body = body.isBlockStatement() ? body.get("body") : [ body ];
+
+        liftDeclaration(body, "var");
+        liftDeclaration(body, "let");
+
+        function liftDeclaration(body, kind) {
```

Comment:
this function will be declared for every ForStatement, move it to an outer scope ?